pkg/awsutils/awsutils.go

```diff
@@ -223,6 +228,10 @@ func New() (*EC2InstanceMetadataCache, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Clean up leaked ENIs in the background
+	go cache.cleanUpLeakedENIs()
```

Comment:
Instead of running a goroutine which infinitely loops with sleep in each iteration, you can do it a bit more cleanly as follows:

```
go wait.Until(cache.cleanUpLeakedENIs, time.Hour, wait.NeverStop)
```

And then the function itself can be a one-off operation:

```
func (cache *EC2InstanceMetadataCache) cleanUpLeakedENIs() {
// list enis
// delete enis
}
```

Reply:
Found this:
```
go wait.Forever(cache.cleanUpLeakedENIs, time.Hour)
```

Reply:
```
2019-09-27T18:59:25.632Z [INFO]	Will attempt to clean up AWS CNI leaked ENIs after waiting 3m1s.
2019-09-27T19:02:26.633Z [DEBUG]	Checking for leaked AWS CNI ENIs.
2019-09-27T19:02:26.934Z [DEBUG]	No AWS CNI leaked ENIs found.
2019-09-27T20:02:26.934Z [INFO]	Will attempt to clean up AWS CNI leaked ENIs after waiting 2m9s.
2019-09-27T20:04:35.934Z [DEBUG]	Checking for leaked AWS CNI ENIs.
2019-09-27T20:04:36.057Z [DEBUG]	No AWS CNI leaked ENIs found.
```